### PR TITLE
Ushakov task 4

### PR DESCRIPTION
**Pull request by AurelVU**

### CODE REVIEW

**Review by DmitryPoliuha — COMMENTED** [review]
Looks good 👍

File: ushakov_va/task_4/olxscrapper/pagescrapper.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            "//li[@class='offer-details__item'][2]/a[@class='offer-details__param offer-details__param--url']/strong[@class='offer-details__value']/text()").get().strip()
+        item['photos_urls'] = [i.get() for i in response.xpath("//li[@class='fleft']/a/@href")]
+        item['state'] = (
+                [i for i in response.xpath("//strong[@class='offer-details__value']/text()") if i.root == 'Б/у'] +
+                [i for i in response.xpath("//strong[@class='offer-details__value']/text()") if i.root == 'Новый'])[
+            0].get()
+        item['description'] = ''.join([i.get().strip() for i in response.xpath("//div[@id='textContent']/text()")])
+        item['date_of_creation'] = datetime.strptime(
+            _multiple_replace(response.xpath("//li[@class='offer-bottombar__item'][1]/em/strong/text()").get().strip(),
+                              _months), 'в %H:%M, %d %b %Y').strftime("%Y-%m-%d-%H.%M.%S")
```
Comment (DmitryPoliuha) [inline ushakov_va/task_4/olxscrapper/pagescrapper.py:80]: not sure if I understand what are you doing here 😅 could you explain, please?

File: ushakov_va/task_4/olxscrapper/pagescrapper.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            yield scrapy.Request(url=absoluteLink, callback=self.parse_item)
+
+    @check_count
+    def parse_item(self, response):
+        item = Product()
+        item['price'] = response.xpath("//div[@class='pricelabel']/strong/text()")[0].get().strip()
+        item['name'] = response.xpath("//div[@class='offer-titlebox']/h1/text()").get().strip()
+        item['category'] = response.xpath(
+            "//li[@class='offer-details__item'][2]/a[@class='offer-details__param offer-details__param--url']/strong[@class='offer-details__value']/text()").get().strip()
+        item['photos_urls'] = [i.get() for i in response.xpath("//li[@class='fleft']/a/@href")]
```
Comment (DmitryPoliuha) [inline ushakov_va/task_4/olxscrapper/pagescrapper.py:72]: doesn't `xpath` have something like `get_all()`?

File: ushakov_va/task_4/olxscrapper/pagescrapper.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        item['state'] = (
+                [i for i in response.xpath("//strong[@class='offer-details__value']/text()") if i.root == 'Б/у'] +
+                [i for i in response.xpath("//strong[@class='offer-details__value']/text()") if i.root == 'Новый'])[
+            0].get()
```
Comment (DmitryPoliuha) [inline ushakov_va/task_4/olxscrapper/pagescrapper.py:76]: what are you doing here?

File: ushakov_va/task_4/olxscrapper/pagescrapper.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            _multiple_replace(response.xpath("//li[@class='offer-bottombar__item'][1]/em/strong/text()").get().strip(),
+                              _months), 'в %H:%M, %d %b %Y').strftime("%Y-%m-%d-%H.%M.%S")
+        item['views_count'] = response.xpath(
+            "//li[@class='offer-bottombar__item'][2]/span[@class='offer-bottombar__counter']/strong/text()").get().strip()
+        item['id'] = response.xpath("//li[@class='offer-bottombar__item'][3]/strong/text()").get().strip()
+        item['url'] = response.url.strip()
+        item['seller_name'] = response.xpath("//div[@class='offer-user__actions']/h4/a/text()").get().strip()
+        item['seller_address'] = response.xpath("//div[@class='offer-user__address']/address/p/text()").get().strip()
+        item['olx_delivery_availability'] = True if len(
+            response.xpath("//span[@class=' AdPageBox__link-button-caption  svelte-14f8vge']")) > 0 else False
```
Comment (DmitryPoliuha) [inline ushakov_va/task_4/olxscrapper/pagescrapper.py:88]: `bool(len(response.xpath("//span[@class=' AdPageBox__link-button-caption  svelte-14f8vge']")))`

File: ushakov_va/task_4/olxscrapper/pagescrapper.py (hunk 1 above)
Comment (AurelVU) [reply ushakov_va/task_4/olxscrapper/pagescrapper.py]: It is fixed now

File: ushakov_va/task_4/olxscrapper/pagescrapper.py (hunk 2 above)
Comment (AurelVU) [reply ushakov_va/task_4/olxscrapper/pagescrapper.py:72]: I dont know. Found this way

File: ushakov_va/task_4/olxscrapper/pagescrapper.py (hunk 3 above)
Comment (AurelVU) [reply ushakov_va/task_4/olxscrapper/pagescrapper.py:76]: On the site, the status field was different in different situations. Here I pull together all the blocks where there may be state data, filter by the value of the text in them (maybe one or the other) and get text from it

File: ushakov_va/task_4/olxscrapper/pagescrapper.py (hunk 4 above)
Comment (AurelVU) [reply ushakov_va/task_4/olxscrapper/pagescrapper.py]: Thanks

**Review by PolyMaG — APPROVED** [review]

**Pull request merged**